### PR TITLE
make finalizeSize() MI-ready

### PR DESCRIPTION
**Pull request by WalterBright**
Did similar change as to membersToDt() to bend the compiler logic to accommodate (eventual) multiple inheritance.

### CODE REVIEW

**Comment by WalterBright** [comment]
The travis build failure seems unrelated.

File: src/dclass.d [hunk 1]
```diff
@@ -1162,19 +1162,63 @@ public:
...
+                        b2 = &b2.baseInterfaces[0];
+                        b2.offset = offset;
+                        //printf("\tvtblInterfaces[%d] b=%p   sym = %s, offset = %d\n", bi, b2, b2.sym.toChars(), b2.offset);
+                    }
+                }
+                membersPlace(b.sym, offset);
+                //printf(" %s size = %d\n", b.sym.toChars(), b.sym.structsize);
+                offset += b.sym.structsize;
+                if (alignsize < b.sym.alignsize)
+                    alignsize = b.sym.alignsize;
```
Comment (andralex) [inline src/dclass.d]: could have used min here... yum, phobos

File: src/dclass.d [hunk 2]
```diff
@@ -1162,19 +1162,63 @@ public:
...
+            return offset - baseOffset;
         }
 
-        // Add vptr's for any interfaces implemented by this class
-        structsize += setBaseInterfaceOffsets(structsize);
+        structsize += membersPlace(this, structsize);
 
         uint offset = structsize;
-        for (size_t i = 0; i < members.dim; i++)
+        foreach (s; *members)
```
Comment (andralex) [inline src/dclass.d]: kewl

**Comment by andralex** [comment]
good to finally see some D style in dmd

**Comment by andralex** [comment]
Auto-merge toggled on

**Pull request merged**